src/app/Models/Product.php

```diff
@@ -46,6 +47,17 @@ public function images()
         return $this->hasMany(Image::class,'product_id');
     }
 
+    public function searchableAs()
+    {
+        return "products";
+    }
+
+    public function toSearchableArray()
+    {
+        $array = $this->toArray();
+        return $array;
+    }
```

Comment:
Ý nghĩa của hàm searchableAs() và toSearchableArray() dùng để làm gì?

Reply:
mỗi model nó sẽ đồng bộ với 1 chỉ mục tìm kiếm index,nó sẽ chứa tất cả các cột để mình tìm kiếm.hiểu theo cách khác có nghĩa là mỗi index là một bảng đó anh.Hàm searchableAs() nó sẽ return về index mình cần tìm kiếm(nó cho ta biết sẽ tìm kiếm ở bảng nào),có thể k cần hàm này thì nó vẫn hiểu đó anh.

Reply:
còn hàm toSearchableArray() dùng để tùy chỉnh cột tìm kiếm,ở đây nó mặc định tìm kiếm toàn bộ,nếu mình thích có thể tùy chỉnh cột tìm kiếm

Reply:
theo em hiểu là nó hoạt động như v